crates/node/src/lib.rs

```diff
@@ -424,6 +438,51 @@ impl Node {
             self.server.run(config.common.clone()),
         )?;
 
+        let my_roles = my_node_config.roles;
+        // Report that the node is running when all roles are ready
+        let _ = tc.spawn(TaskKind::Disposable, "status-report", None, async move {
+            self.health
+                .node_status()
+                .wait_for_value(NodeStatus::Alive)
+                .await;
+            trace!("Node-to-node networking is ready");
+            for role in my_roles {
+                match role {
+                    Role::Worker => {
+                        self.health
+                            .worker_status()
+                            .wait_for_value(WorkerStatus::Ready)
+                            .await;
+                        trace!("Worker role is reporting ready");
+                    }
+                    Role::Admin => {
+                        self.health
+                            .admin_status()
+                            .wait_for_value(AdminStatus::Ready)
+                            .await;
+                        trace!("Worker role is reporting ready");
+                    }
+                    Role::MetadataStore => {
+                        self.health
+                            .metadata_server_status()
+                            .wait_for_value(MetadataServerStatus::Ready)
+                            .await;
+                        trace!("Metadata role is reporting ready");
+                    }
+
+                    Role::LogServer => {
+                        self.health
+                            .log_server_status()
+                            .wait_for_value(LogServerStatus::Ready)
+                            .await;
+                        trace!("Log-server is reporting ready");
+                    }
+                }
+            }
+            tracing::info!("Restate server is ready");
```

Comment:
Why using `tracing` prefix?

Reply:
No particular reason. Let me remove that.